pkg/gitreceive/storage/client.go

```diff
@@ -18,10 +21,10 @@ func GetClient(regionStr string) (*s3.S3, error) {
 		return nil, err
 	}
 
-	return s3.New(session.New(&aws.Config{
-		Credentials:      auth,
-		Region:           aws.String(regionStr),
-		Endpoint:         aws.String(endpoint),
-		S3ForcePathStyle: aws.Bool(true),
-	})), nil
+	// the New function call guesses which signature version to use. Currently, it correctly guesses V2 for GCS and V4 for both AWS S3 and Minio
+	s3Client, err := s3.New(endpoint.URLStr, auth.accessKeyID, auth.accessKeySecret, !endpoint.Secure)
+	if err != nil {
```

Comment:
client signature should change according to the storage back end we are accessing . For minio it should be s3.NewV4. for other s3.NewV2